manifests/init.pp

```diff
@@ -140,38 +144,39 @@
-  Simplib::Host                          $srv_domain           = $facts['domain'],
-  Stdlib::AbsolutePath                   $ssldir               = $::pupmod::params::puppet_config['ssldir'],
-  Simplib::Syslog::Facility              $syslogfacility       = 'local6',
-  Boolean                                $use_srv_records      = false,
-  Stdlib::AbsolutePath                   $vardir               = $::pupmod::params::puppet_config['vardir'],
-  Boolean                                $haveged              = simplib::lookup('simp_options::haveged', { 'default_value' => false }),
-  Boolean                                $fips                 = simplib::lookup('simp_options::fips', { 'default_value' => false }),
-  Boolean                                $firewall             = simplib::lookup('simp_options::firewall', { 'default_value' => false }),
-  Hash                                   $pe_classlist         = {},
-  String[1]                              $package_ensure       = simplib::lookup('simp_options::package_ensure' , { 'default_value' => 'installed'}),
-  Boolean                                $set_environment      = true,
-  Boolean                                $mock                 = false
+  Variant[Simplib::Host,Enum['$server']] $ca_server                    = simplib::lookup('simp_options::puppet::ca', { 'default_value' => '$server' }),
+  Simplib::Port                          $ca_port                      = simplib::lookup('simp_options::puppet::ca_port', { 'default_value' => 8141 }),
+  Simplib::Host                          $puppet_server                = simplib::lookup('simp_options::puppet::server', { 'default_value' => "puppet.${facts['domain']}" }),
+  Simplib::ServerDistribution            $server_distribution          = simplib::lookup('simp_options::puppet::server_distribution', { 'default_value' => 'PC1' } ),
+  Optional                               $ca_crl_pull_interval         = undef,
+  Simplib::Host                          $certname                     = $facts['fqdn'],
+  String[0]                              $classfile                    = '$vardir/classes.txt',
+  Stdlib::AbsolutePath                   $confdir                      = $::pupmod::params::puppet_config['confdir'],
+  Boolean                                $daemonize                    = false,
+  Enum['md5','sha256']                   $digest_algorithm             = 'sha256',
+  Boolean                                $enable_puppet_master         = false,
+  Stdlib::AbsolutePath                   $environmentpath              = $::pupmod::params::puppet_config['environmentpath'],
+  Boolean                                $listen                       = false,
+  Stdlib::AbsolutePath                   $logdir                       = $::pupmod::params::puppet_config['logdir'],
+  Simplib::Port                          $masterport                   = 8140,
+  Boolean                                $remove_environment_from_main = true,
```

Comment:
@pillarsdotnet I see what you're getting at here and agree with the concept but I'm not super keen on adding a 'workaround' variable to the API.

What do you think about instead changing `$set_environment` to the following:
* `true` => Set the environment and clean the environment out of `main`
* `no_clean` => Set the environment and do not clean the environment out of `main`
* `false` => Don't do anything